lib/resty/websocket/client.lua

```diff
@@ -98,9 +98,17 @@ function _M.connect(self, uri, opts)
         path = "/"
     end
 
-    local ssl_verify, proto_header, origin_header, sock_opts = false
+    local ssl_verify, proto_header, origin_header, custom_header, sock_opts = false
 
     if opts then
+        local headers = opts.headers
+        if headers then
+            custom_header = ""
+            for field,value in pairs(headers) do
+                custom_header = custom_header .. "\r\n" .. field .. ": " .. value
```

Comment:
It's very bad to use `..` in a loop. Lua strings are immutable and this can easily lead to `O(n^2)` time complexity and `O(n^2)` space complexity. Better use a globally shared Lua table to collect all these string pieces.